utils/mil_tools/nodes/label_me_bag.py

```diff
@@ -0,0 +1,231 @@
+                  raise Exception("Segment does not contain 'start' attribute {}".format(segment))
+              if not 'stop' in segment:
+                  raise Exception("Segment does not contain 'stop' attribute {}".format(segment))
+              if not 'topic' in segment:
+                  raise Exception("Segment does not contain 'topic' attribute {}".format(segment))
+              if not 'name' in segment:
+                  raise Exception("Segment does not contain 'name' attribute {}".format(segment))
+              if not 'freq' in segment:
+                  raise Exception("Segment does not contain 'freq' attribute {}".format(segment))
+
+  def _read_bag(self, bagfile, segments):
+      bag = rosbag.Bag(bagfile)
+      _, _, first_time = bag.read_messages().next()
+      for segment in segments:
+          print "\tProccessing Segment '{}' with topic '{}'".format(segment['name'], segment['topic'])
+          path = os.path.join(self.labelme_dir, 'Images', segment['name'], segment['topic'].replace('/', '@'))
+          if not os.path.exists(path):
+              os.makedirs(path)
+          start = first_time + rospy.Duration(segment['start'])
+          stop = first_time + rospy.Duration(segment['stop'])
+          interval = rospy.Duration(1.0 / segment['freq'])
+          print "\t\tSaving Images from duration {} to {} every {} seconds".format(segment['start'], segment['stop'], interval.to_sec())
+          next_time = start
+          for topic, msg, time in bag.read_messages(topics=[segment['topic']], start_time = start, end_time = stop):
+              if time >= next_time:
+                  img = self.bridge.imgmsg_to_cv2(msg, desired_encoding="passthrough")
+                  filename = os.path.join(path, str(msg.header.stamp) + '.jpg')
+                  cv2.imwrite(filename, img)
+                  next_time = next_time + interval
+      
+  def _print_bag_report(self, bagconfig):
+      total_xml_count = 0
+      total_img_count = 0
+      for segment in bagconfig['segments']:
+          print "\t\tGenerating Report for segment '{}' with topic '{}'".format(segment['name'], segment['topic'])
+          xml_path = os.path.join(self.labelme_dir, 'Annotations', segment['name'], segment['topic'].replace('/', '@'))
+          img_path = os.path.join(self.labelme_dir, 'Images', segment['name'], segment['topic'].replace('/', '@'))
+          xml_count = 0
+          img_count = 0
+          for xmlfile in os.listdir(xml_path):
+              xml_count += 1
+          for imgfile in os.listdir(img_path):
+              img_count += 1
+          print "\t\t\t{}/{} images labeled in this segment ({}%)".format(xml_count, img_count, round(100.0*xml_count/img_count,2))
+          total_xml_count += xml_count
+          total_img_count += img_count
+      print "\t\t{}/{} images labeled in this bag ({}%)".format(total_xml_count, total_img_count, round(100.0*total_xml_count/total_img_count,2))
```

Comment:
You do the percent function a lot, maybe break it out somewhere to clean up code?